client/packages/system/src/Item/Components/StockItemSearchInput/StockItemSearchInput.tsx

```diff
@@ -26,7 +31,16 @@ export const StockItemSearchInput: FC<StockItemSearchInputProps> = ({
   includeNonVisibleWithStockOnHand = false,
   itemCategoryName,
 }) => {
-  const { filter, onFilter } = useStringFilter('name');
+  const { filter, onFilter } = useStringFilter('codeOrName');
+  const [search, setSearch] = useState('');
+
+  const [selectedCode, setSelectedCode] = useState('');
+
+  const debounceOnFilter = useDebouncedValueCallback(
```

Comment:
switched to this useDebouncedValueCallback - useDebounceCallback just fired every call after a delay, rather than only the most recent one 🤷‍♀️